### PR TITLE
feat: allow to host dbt docs on cluster

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR allows hosting dbt docs locally and on k8s. It also stores the `manifest.json` state file and superset exposures in clickhouse enabling us to remove the volume for the state. Closes: https://github.com/openedx/tutor-contrib-aspects/issues/915

- Fixes an issue with the dbt state that makes it find false positives.
- Turn every rendered value into an env var.
- Removes the k8s volume for the dbt state in favor of storing metadata in ClickHouse.

### Testing instructions

- Enable `RUN_ASPECTS_DOCS` in your `config.yml`.
- Run migrations with `tutor dev do alembic -c "upgrade head"`
- Build the aspects docker image: `tutor images build aspects`
- Run DBT.
- Run collect lineage command.
- Go to http://dbt.local.edly.io:7000 and verify the docs are working, lineage data is available and exposures are available.
- Rerun DBT and verify that no changes have been detected.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: tutoraspects/patches/local-docker-compose-jobs-services [hunk 1]
```diff
@@ -2,7 +2,13 @@ aspects-job:
...
   image: {{ DOCKER_IMAGE_ASPECTS }}
   environment:
-    - XAPI_SCHEMA={{ ASPECTS_XAPI_DATABASE }}
+    ASPECTS_EVENT_SINK_DATABASE: {{ASPECTS_EVENT_SINK_DATABASE}}
+    ASPECTS_XAPI_DATABASE: {{ASPECTS_XAPI_DATABASE}}
+    CLICKHOUSE_CLUSTER_NAME: {{CLICKHOUSE_CLUSTER_NAME}}
+    DBT_STATE: {{ DBT_STATE_DIR }}
+    ASPECTS_DATA_TTL_EXPRESSION: "{{ ASPECTS_DATA_TTL_EXPRESSION }}"
+    DBT_PROFILE_TARGET_DATABASE: "{{ DBT_PROFILE_TARGET_DATABASE }}"
+    DBT_PROFILES_DIR: /app/aspects/dbt/
```
Comment (saraburns1) [inline tutoraspects/patches/local-docker-compose-jobs-services:11]: should we make this an env variable?

Comment (Ian2012) [reply tutoraspects/patches/local-docker-compose-jobs-services:11]: Probably

**Comment by itsjeyd** [comment]
Hey @bmtcril and @pomegranited, would you have time to give this PR a look?

File: tutoraspects/templates/aspects/build/aspects/entrypoint [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+
+if [ -z "${DBT_SSH_KEY+x}"]
```
Comment (bmtcril) [inline tutoraspects/templates/aspects/build/aspects/entrypoint:8]: I think you need a space before the "]". I'm seeing:

`/entrypoint: line 8: [: missing `]'`

File: tutoraspects/templates/aspects/build/aspects/entrypoint [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+fi
+
+rm -rf aspects-dbt
+
+echo "Installing aspects-dbt"
+echo "git clone -b ${DBT_BRANCH} ${DBT_REPOSITORY}"
+git clone -b ${DBT_BRANCH} ${DBT_REPOSITORY} aspects-dbt
+
+cd aspects-dbt
+
```
Comment (bmtcril) [inline tutoraspects/templates/aspects/build/aspects/entrypoint:26]: To get the exposures you'll need to add the dbt-state volume to this service like we do in local-docker-compose-jobs-service and then just copy the generated file in (but don't fail if it doesn't exist) with something like:

```
echo "Trying to copy the Superset exposures file generated by the collect-dbt-lineage command"
cp /app/aspects/dbt_state/superset_exposures.yaml ./models || :
```

Comment (Ian2012) [reply tutoraspects/templates/aspects/build/aspects/entrypoint:26]: I'm storing those files on ClickHouse but that's WIP. It's a small refactor but requires some time to finish.

File: tutoraspects/patches/local-docker-compose-services [hunk 4]
```diff
@@ -123,3 +121,31 @@ aspects-consumer:
...
+    DBT_BRANCH: "{{ DBT_BRANCH }}"
+    DBT_REPOSITORY: "{{ DBT_REPOSITORY }}"
+    DBT_SSH_KEY: "{{ DBT_SSH_KEY }}"
+  command: |
+    bash /app/aspects/scripts/docs.sh
+  ports:
+    - 7000:7000
+  volumes:
+    - ../../env/plugins/aspects/apps/aspects/dbt:/app/aspects/dbt
+    - ../../env/plugins/aspects/apps/aspects/scripts:/app/aspects/scripts
```
Comment (Ian2012) [inline tutoraspects/patches/local-docker-compose-services:147]: reminder: this volume is duplicated, and needs to be removed.

**Comment by bmtcril** [comment]
Seems a little closer, I can init now, but k8s complains about the format of the deployment:

```
Error from server (BadRequest): error when creating "/Users/brianmesick/Dev/oars-tutor/env": Deployment in version "v1" cannot be handled as a Deployment: strict decoding error: unknown field "spec.template.metadata.containers", unknown field "spec.template.metadata.volumes"
Error: Command failed with status 1: kubectl apply --kustomize /Users/brianmesick/Dev/oars-tutor/env --selector app.kubernetes.io/component notin (job,volume,namespace)
```

File: tutoraspects/patches/local-docker-compose-services (hunk 4 above)
Comment (bmtcril) [reply tutoraspects/patches/local-docker-compose-services:147]: Does this still need to be done?

**Review by bmtcril — COMMENTED** [review]
Looks like the code is in a good place! Can you document the new feature and setting, why people might want it, and why they should be careful about turning it on / what access controls people might want? A "how-to" in openedx-aspects is probably the best place for it.

**Review by bmtcril — APPROVED** [review]
Tested in k8s and local and is working well

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-aspects&entry.867055334=https://github.com/openedx/tutor-contrib-aspects/pull/892&entry.1484655318=https://github.com/Ian2012&entry.752974735=2024-07-12+14:52&entry.1917517419=2024-08-20+16:31&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**